dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -1180,17 +1182,37 @@
 	</PropertyGroup>
 
 	<Target Name="_ComputeDynamicLibrariesToReidentify">
-		<PropertyGroup>
-			<_ExecutablePathPrefix Condition="'$(_PlatformName)' == 'iOS' Or '$(_PlatformName)' == 'tvOS'">@executable_path/</_ExecutablePathPrefix>
-			<_ExecutablePathPrefix Condition="'$(_PlatformName)' == 'macOS' Or '$(_PlatformName)' == 'MacCatalyst'">@executable_path/../$(_CustomBundleName)/</_ExecutablePathPrefix>
+		<!-- iOS and tvOS -->
+		<PropertyGroup Condition="'$(_PlatformName)' == 'iOS' Or '$(_PlatformName)' == 'tvOS'">
+			<_ExecutablePathPrefix>@executable_path/</_ExecutablePathPrefix>
 		</PropertyGroup>
-		<ItemGroup>
+		<ItemGroup Condition="'$(_PlatformName)' == 'iOS' Or '$(_PlatformName)' == 'tvOS'">
 			<_DynamicLibraryToReidentify Include="@(_FileNativeReference)" Condition="'%(Extension)' == '.dylib'">
-				<DynamicLibraryId>$(_ExecutablePathPrefix)%(Filename)%(Extension)</DynamicLibraryId>
 				<SourceItemGroup>_FileNativeReference</SourceItemGroup>
+				<DynamicLibraryId>$(_ExecutablePathPrefix)%(Filename)%(Extension)</DynamicLibraryId>
 				<ReidentifiedPath>$(_IntermediateNativeLibraryDir)/%(Filename)%(Extension)</ReidentifiedPath>
 			</_DynamicLibraryToReidentify>
 		</ItemGroup>
+		<!-- macOS and MacCatalyst: shared libraries might be deployed in subdirectories and RelativePath must be respected -->
+		<PropertyGroup Condition="'$(_PlatformName)' == 'macOS' Or '$(_PlatformName)' == 'MacCatalyst'">
+			<_ExecutablePathPrefix >@executable_path/../../</_ExecutablePathPrefix>
+		</PropertyGroup>
+		<ItemGroup Condition="'$(_PlatformName)' == 'macOS' Or '$(_PlatformName)' == 'MacCatalyst'">
```

Comment:
While iOS and tvOS might not need these changes, they should still work for both platforms - so that makes me think it's better to keep the code identical for all platforms (easier to maintain and test).

Reply:
I have merged everything back and kept everything the same for both platforms